include/deal.II/trilinos/nox.templates.h

```diff
@@ -940,9 +940,17 @@ namespace TrilinosWrappers
     , n_jacobian_applications(0)
     , n_nonlinear_iterations(0)
     , n_last_linear_iterations(0)
+    , pending_exception(nullptr)
   {}
 
 
+  template <typename VectorType>
+  NOXSolver<VectorType>::~NOXSolver()
+  {
+    Assert(pending_exception == nullptr, ExcInternalError());
```

Comment:
```suggestion
    AssertNothrow(pending_exception == nullptr, ExcInternalError());
```

Reply:
I'd like to keep it as an `Assert`. The logic in the file is so that there *shouldn't* be an exception left. If there is, there's a programming bug; it isn't a possible run-time error.

Reply:
That's a reasonable point, lets keep it as an assert.